### PR TITLE
Maxwells fun models test

### PR DESCRIPTION
**Pull request by maxwellherron5**
Bit of a messy PR because I jumped around on a couple different things, but here are the cliff notes

- Added a utility to our data_loader module to actually build the pytorch data loader objects along with retaining the one hot encoding information so we can use that to decode predictions during inference
- Created a super basic MLP to act as a PoC for how our real models will be developed and run against the chroma/lab data
- Created a solver class that can handle generic model utilities so we don't need to reinvent the wheel for each model

MUCH needs to be done here still, tbh this PR really is just a PoC but I think it gets us in a good place for experimentation. Next step here is updating the data pipeline so that our models can train on any number of associated chroma/lab files at a time and not just one.

### CODE REVIEW

File: solver.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+            self.train_accuracy_history.append(train_accuracy)
+            self.valid_accuracy_history.append(val_accuracy)
+
+            print(
+                f"Training Loss: {avg_train_loss:.4f}. Validation Loss: {avg_val_loss:.4f}."
+            )
+            print(
+                f"Training Accuracy: {train_accuracy:.4f}. Validation Accuracy: {val_accuracy:.4f}."
+            )
```
Comment (maxwellherron5) [inline solver.py:103]: Look familiar? 😉

File: models/mlp_chord_classifier.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        self.relu2 = nn.ReLU()
+        self.fc3 = nn.Linear(32, num_classes)
+
+    def forward(self, x):
+        x = self.fc1(x)
+        x = self.relu1(x)
+        x = self.fc2(x)
+        x = self.relu2(x)
+        x = self.fc3(x)
+        return x
```
Comment (maxwellherron5) [inline models/mlp_chord_classifier.py:19]: OpenAI can feel free to hire me whenever for this masterpiece

File: models/mlp_chord_classifier.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+            nn.Linear(input_size, 64),
+            nn.ReLU(),
+            nn.Linear(64, 32),
+            nn.ReLU(),
+            nn.Linear(32, num_classes)
+        )
+
+    def forward(self, x):
+        x = self.network(x)
+        return x
```
Comment (maxwellherron5) [inline models/mlp_chord_classifier.py:17]: OpenAI can hire me whenever for this masterpiece

File: main.py
Comment (maxwellherron5) [inline main.py:1]: Still some hardcoded pieces that will need to be adjusted as needed for local testing, but this is the basis for running via a config file. Note the FIXME comment, I need to resolve that issue but I'm too eepy to think more about it at the moment

File: solver.py [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+        self.epochs = kwargs.pop("epochs", 10)
+
+        if model:
+            self.model = model.to(self.device)
+        else:
+            match self.model_type:
+                case "MLPChordClassifier":
+                    self.model = MLPChordClassifier(**self.model_kwargs)
+                case _:
+                    self.model = MLPChordClassifier(**self.model_kwargs)
```
Comment (maxwellherron5) [inline solver.py:32]: Yes I realize how silly this is lol but that's only because we only have one model type going into main at the moment :joy: as you create new models, you will need to update the match statement here to allow for them

File: configs/default_config.yaml
Comment (maxwellherron5) [inline configs/default_config.yaml:1]: Super basic for now. Moving forward I'm gonna update the config to allow us to specify scheduler/criterion/optimizer and their params as well

**Review by zhutwo — APPROVED** [review]
is gud
